### DISCUSSION TITLE
@vedgy is a hypocrite.

### DISCUSSION THREAD

**Discussion by shenlebantongying** [goldendict/goldendict#1619] [General]
See how this "maintainer:clown_face:" discuss things with the biggest cancer of this repo https://github.com/goldendict/goldendict/issues/1424 instead of reviewing a PR https://github.com/goldendict/goldendict/issues/1625

---

He claims xiaoyifang's fork contains unrelated changes, and yet in his PR there are tons of unrelated changes, because introducing the new WebEngine requires changing related code. Some obvious bugs will be just "accidentally" fixed.

This may indicate that he don't even try to think about what is needed for the webengine migration when he reviews the code.

He keeps saying "This formatting is wrong", and yet in this repo, there are 4 styles of `if`

```
if(){
if () {
if()
{
if () {
```

Plus two styles of `else` placement

```
} else {
}
else {}
```

Plus `if` without brackets Plus 2 & 4 spaces indents.

More than 10 styles of `if`! STFU, if there is no standard at all. The problem is solved in xiaoyifang's fork with .clang-format and all new code are uniformed.

He keeps saying that one QMutex's problem, but yet the entire codebase contains many useless locks that should be removed. Understandable due to the rapid development 10 years ago, but if he doesn't know that many locks are useless, he probably doesn't even understand how data are processed after looking at the code for years.

There are 500+ issues opened there and many easy PR waiting to be merged. And yet he refuse to admit the new fork fixed a lot of them.

Some are extremely easy to fix. Just someone, anyone read the issue, and change a few lines, then someone presses the "Merge" button, and the issue is fixed. And Yet, those 500 sit here for years and NOTHING happens.

### Accepting his PR is Harmful, thus it goes nowhere.

Can you image when google forked webkit, they keep a conditional compile of `webkit` mode?

5k and more lines trying to blend WebKit & WebEngine. Linux's distros have either removed or have agenda to remove QtWebkit. On Windows & macOS, there is no official binary available at all.

No one other than him can even compile the code!

Who do u expect to fix bugs on this monstrosity? Merging this will make no one be able to work on this.

He keeps suggesting moving some build scripts changes to another PR, but yet this project cannot even be built on Windows & macOS anymore.

He says, "xiaoyifang apparently never bothered to compile a Qt WebKit version of GoldenDict and see how it works and what behavior should be preserved."

What is shit statement? WebEngine is a fucking webview, what do u expect? It displays web pages!

Right now, the WebEngine is not event the biggest change of xiaoyifang's fork. For example, the new extremely fast full text search. The old homemade fts indexer will turn any old hardware to a jet engine if users have too many dictionaries.

---

He is aware that big changes are not going to be merged since some of his own PR is not merged.

And yet, he torturing others like no one else. Why waste his entire day just for nothing??? https://github.com/goldendict/goldendict/pull/1515

If you tell xiaoyifang that the big PR is not going to be merged, he can waste less time.

![Untitled](https://user-images.githubusercontent.com/20123683/233199313-3c3caefa-f14f-4e6a-a940-3641a4fee267.png)

---

@vedgy cannot resist his ego devouring himself. Just because the code changes are not made by him, he won't even accept them.

@vedgy is in a narcissistic hypocrite who still holding his unfair judgement from last year.

@vedgy is one of the gatekeepers in the free software community who tries to keep newcomers, usually less experienced ones, from joining in. What's wrong with experimenting? 1st versions of many important software today are buggy garbage.

@vedgy should apologize for his unfair judgements. Of course, he doesn't have to. Just stop defaming like a :clown_face:, STFU, and stop his stupid attempt and stop this useless war of talking.

**_!! Stop talking as if you are a maintainer if YOU DON'T EVEN HAVE COMMIT PERMISSION! !!_**

**_!! Stop acting as if you are a maintainer if YOU DON'T EVEN HAVE COMMIT PERMISSION! !!_**

I am aware that you are also kdevelop's developer. Why don't u stop worrying about this repo? Just enjoy this repo's HEAD built on your computer and focus on your other things??

**Comment by vedgy** [comment]
I wrote the "rich source of ideas" comment while I was still working on the initial version of #1542. Subsequently I had to discard some of the ideas and replace them with better alternatives. But I still mostly stand by that old usefulness statement, because @xiaoyifang's experimental porting allowed to estimate the effort necessary to port properly. The "prototype" and "rich source of ideas" formulations in no way contradict the "I had to implement almost everything differently, because there were no clean, bug-free commits to borrow." statement.

@shenlebantongying has deliberately flooded #1618 and this personal attack "discussion" with numerous false, half-true, misleading, incoherent and off-topic statements. The comments [s]he has posted reveal the attitude and character of this person to any open-minded observer. @shenlebantongying is clearly a troll and a parasite on GoldenDict repository. [S]he has contributed nothing useful and clearly deserves to be blocked. But I am still glad to have started #1618: at least these trolling comments don't derail some unlucky issues and pull requests.

**Reply by shenlebantongying** [reply]
> deliberately flooded

Yes, after you trying to shut me up after I post a single "This is fixed" message.

WHY ARE YOU TRYING TO STOP ME from POSTING SIMPLE MESSAGES??

YOU BRING THAT SINGLE MESSAGE TO A WHOLE NEW LEVEL.

In fact, I only discuss technical matters and avoid all trolling conversations coming from the REAL CANCER of this repo before this week and do not care about political battles.

The fact you call me out before the one who have been sabotaging this project is ridiculous.

Plus you post malicious post with outdated misinformation https://github.com/goldendict/goldendict/discussions/1616

> contributed nothing useful and clearly deserves to be blocked. 

Please ban me from this repo, I don't care, but sadly you don't even have permission to do it :clown_face:

**Reply by vedgy** [reply]
I don't have motivation to reply further to such statements. Anyone interested can judge communication in #1616, #1618 and this "discussion" for him- or herself.

**Reply by shenlebantongying** [reply]
Ban someone only need a few clicks, and yet there is a cancer trolling for crazily long time.

The real problem is how it comes to here.

Isn't it because the maintainership of this repo is useless?
